yodeploy/tests/test_application.py

```diff
@@ -36,10 +40,29 @@ class AttrDict(dict):
         store = LocalRepositoryStore(self.mkdir('artifacts'))
         self.repo = Repository(store)
         self.app = Application('test', self.tmppath('config.py'))
-        if not os.path.exists('test-data/deploy-ve/virtualenv.tar.gz'):
+
+        self.test_ve_tar_path = self._data_path('deploy-ve/virtualenv.tar.gz')
+        self.test_req_path = self._data_path('deploy-ve/requirements.txt')
+        self.test_ve_path = self._data_path('deploy-ve')
+
+        if not os.path.exists(self.test_ve_tar_path):
             self._create_test_ve()
-        self._deploy_ve_hash = ve_version(sha224sum(
-            'test-data/deploy-ve/requirements.txt'))
+        self._deploy_ve_hash = ve_version(sha224sum(self.test_req_path))
+
+    def _data_path(self, fragment):
+        version_suffix = 'py%s' % platform.python_version()
+        return os.path.join('test-data-%s' % version_suffix, fragment)
+
+    def _prep_wip_yodeploy_for_install(self):
+        """Copy the yodeploy checkout under test to a temp directory.
+
+        Returns the full path of the yodeploy copy.
+        """
```

Comment:
We do. And, as far as I can tell, easy_install just appends `[easy_install]` to it if it isn't there. It probably also writes out any settings that were passed as parameters.

We can add that section to setup.cfg ourselves to avoid the dirty state. The temp directory that easy_install creates at least look as though it's trying to make a unique path every time (e.g., `/Users/thomas/Sites/foo/egg-dist-tmp-v2m8qX`), so it's unlikely there'd be collisions...

Re-testing that approach confirms some of the above... `easy_install` adds the index-url we're specifying and cleans up our spacing for us.

```diff
diff --git a/setup.cfg b/setup.cfg
index 86d0339..83d033b 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -1,2 +1,6 @@
 [nosetests]
-with-specplugin=1
+with-specplugin = 1
+
+[easy_install]
+index-url = https://<snip>@yolapi.yola.net/simple/
+
```

It's... uncomfortable.